greeneeds/projects/views.py

```diff
@@ -1,3 +1,59 @@
-from django.shortcuts import render
+from unittest           import result
+from django.views       import View
+from django.http        import JsonResponse
 
-# Create your views here.
+from projects.models    import Project, Tag, ProjectImage
+from users.models       import Like
+from django.db.models   import Count
+from datetime           import datetime
+
+class ProjectListView(View):
+    def get(self, request):
+        try:
+            order_condition = request.GET.get('order', None)
+            
+            if order_condition == 'likes':
+                projects = Project.objects.annotate(like_count=Count('Like')).order_by('-like_count')
+            if order_condition == 'recent':
+                projects = Project.objects.order_by('end_datetime')
+            if order_condition == 'random':
+                projects = Project.objects.order_by('?') 
```

Comment:
코드 내용을 보면 프론트 query parameter 를 받아서 분기처릴 하는 것으로 보입니다.
지금은 하나의 key 'order'에 의해 분기가 처리 되므로 이와 같은 상황에서는 dictionary를 사용하여 조금 더 가독성 있는 코드를 작성할 수 있습니다. 
```suggestion
            order_set = {
                'like'   : '-like_count',
                'recent' : 'end_datetime',
                'random  : '?'
            }
            
            order_field = order_set.get(order_condition, '?')
          
            projects = Project.objects.annotate(like_count=Count('Like')).order_by(order_field) 
```